### PR TITLE
Don't allow "required crew" to be less than 1.

### PR DESCRIPTION
**Pull request by quyykk**
**Bugfix:** This PR addresses issue #6578

## Fix Details

`"required crew"` could be reduced to 0 by buying an android. This means that when selling the android a crew member is hired but that is unnecessary since the player exists, which results in an extra crew being hired. The easiest way to fix this is to disallow `"required crew"` to be less than 1.

## Testing Done

Tested buying and selling androids on ships.

### CODE REVIEW

**Comment by Amazinite** [comment]
How does this interact with automata ships? I believe this may make it so that once an automata ship has a command center added to it making it pilotable, it won't be able to be removed.

**Comment by quyykk** [comment]
Very good point! I didn't think of automatons. Anyways, I pushed a fix but haven't tested it with automatons yet.

**Comment by quyykk** [comment]
> I believe this may make it so that once an automata ship has a command center added to it making it pilotable, it won't be able to be removed

Works now as expected.

File: source/Outfit.cpp [hunk 1]
```diff
@@ -346,6 +347,11 @@ int Outfit::CanAdd(const Outfit &other, int count) const
 			if(!minimum)
 				continue;
 		}
+
+		// If this is a automaton then "required crew" can be 0, but not otherwise.
```
Comment (tehhowch) [inline source/Outfit.cpp:351]:
```suggestion
		// Only automatons may have a "required crew" of 0.
```

**Review by tehhowch — COMMENTED** [review]
Comment phrasing, otherwise lgtm.

**Pull request merged**